crates/common/tedge_utils/src/notify.rs

```diff
@@ -65,7 +65,7 @@ pub enum NotifyStreamError {
 }
 
 pub struct NotifyStream {
-    debouncer: debouncer::Debouncer<INotifyWatcher, debouncer::NoCache>,
+    debouncer: debouncer::Debouncer<RecommendedWatcher, debouncer::NoCache>,
```

Comment:
I don't really see how this can help as [the latter is a type alias of the former](https://docs.rs/notify/latest/notify/type.RecommendedWatcher.html), but this is clearer and less dependent on `notify` internals.

Reply:
It helps as if you explicitly put the `INotifiyWatcher` then you have to manually use the `#[cfg(target_os = "linux")]` block, and then do your own logic/research to see which notifier is supported on different target os's. The RecommendedWatcher just allows thin-edge.io to take care of the underlying target_os specific types.

e.g..

```
/// The recommended `Watcher` implementation for the current platform
#[cfg(any(target_os = "linux", target_os = "android"))]
pub type RecommendedWatcher = INotifyWatcher;
/// The recommended `Watcher` implementation for the current platform
#[cfg(all(target_os = "macos", not(feature = "macos_kqueue")))]
pub type RecommendedWatcher = FsEventWatcher;
/// The recommended `Watcher` implementation for the current platform
#[cfg(target_os = "windows")]
pub type RecommendedWatcher = ReadDirectoryChangesWatcher;
/// The recommended `Watcher` implementation for the current platform
#[cfg(any(
    target_os = "freebsd",
    target_os = "openbsd",
    target_os = "netbsd",
    target_os = "dragonflybsd",
    all(target_os = "macos", feature = "macos_kqueue")
))]
```

Reply:
MacOS and freeBSD do not support iNotify, so compiling on those systems would throw an error.